sdks/go/pkg/beam/core/metrics/metrics.go

```diff
@@ -482,7 +482,39 @@ func (mr Results) AllMetrics() QueryResults {
 	return QueryResults{mr.counters, mr.distributions, mr.gauges}
 }
 
-// TODO(BEAM-11217): Implement Query(Filter) and metrics filtering
+// SingleResult interface facilitates metrics query filtering methods.
+type SingleResult interface {
+	Step() string
```

Comment:
Lets not implement Step filtering for now.

At present, I don't think we do a meaningful translation from the internal/uniqueid representation of the transform id, to something the user would be interested in. Until we do, we'd be locked into bad/weird behavior if we change it.

Also "Step" isn't part of the user facing beam model, so it's at best confusing.

At best, please add a comment `// TODO(BEAM-11217): Implement querying metrics by DoFn`
We can figure this out when we also return PCollection metrics, which are also easier to understand by querying by parent DoFn.